src/main/java/frc/robot/controllers/Drivetrain.java

```diff
@@ -1,34 +1,29 @@
 package frc.robot.controllers;
 
-import com.revrobotics.CANSparkMax;
-import com.revrobotics.CANSparkMaxLowLevel.MotorType;
-
 import frc.robot.util.*;
 
-public class Drivetrain
-{
+public abstract class Drivetrain {
     public final double speedFac = -1, turnFac = -1;
 
     public double startPosLeft = 0, startPosRight = 0;
     public double pastLeftDist = 0, pastRightDist = 0;
     public long pastTime;
 
-    public CANSparkMax leftMotor1, leftMotor2, rightMotor1, rightMotor2;
-
-    public PID leftDrivePID = new PID(0.09, 0, 0);
-    public PID rightDrivePID = new PID(0.09, 0, 0);
+    private PID leftDrivePID, rightDrivePID;
     
-    public Drivetrain ()
-    {
-        leftMotor1 = new CANSparkMax(Context.leftMotor1ID, MotorType.kBrushless);
-        leftMotor2 = new CANSparkMax(Context.leftMotor2ID, MotorType.kBrushless);
-        rightMotor1 = new CANSparkMax(Context.rightMotor1ID, MotorType.kBrushless);
-        rightMotor2 = new CANSparkMax(Context.rightMotor2ID, MotorType.kBrushless);
-
+    /**
+     * Requires PIDs to be inputted so each drivetrain can have it's own values
+     * @param leftDrivePID_
+     * @param rightDrivePID_
+     */
+    public Drivetrain(PID leftDrivePID_, PID rightDrivePID_) {
         pastTime = System.currentTimeMillis();
+
+        leftDrivePID = leftDrivePID_;
```

Comment:
Cosmetic, instead of using underscores you could also declare parameters with the same name and then assign them like this:
```java
this.leftDrivePID = leftDrivePID;
```
Both variables end up referencing the same object, assignment statement with "this" just highlights the temporary difference.